src/test/analyserwaveformtest.cpp

```diff
@@ -57,7 +58,9 @@ class AnalyzerWaveformTest : public MixxxTest {
 
 // Basic test to make sure we don't alter the input buffer and don't step out of bounds.
 TEST_F(AnalyzerWaveformTest, canary) {
-    m_aw.initialize(m_pTrack, m_pTrack->getSampleRate(), kBigBufSize);
+    m_aw.initialize(m_pTrack,
+            mixxx::audio::SampleRate(44100),
+            kBigBufSize / kChannelCount);
```

Comment:
This is a test using the artificial canary buffer.
It can be done with any sample rate and will not break. Now with kCanarySampleRate the info is matches the other definitions regarding that buffer. 

Do you have a different solution in mind?

Reply:
> Do you have a different solution in mind?

Just revert to using `pTrack->getSampleRate()` again? There is no reason to decouple these as far as I can tell.